src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -689,6 +708,163 @@ MarkNodeHasMetadata(char *nodeName, int32 nodePort, bool hasMetadata)
 
 
 /*
+ * SequenceDDLCommandsForTable returns a list of commands which create sequences (and
+ * their schemas) to run on workers before creating the relation. The sequence creation
+ * commands are wrapped with a `worker_apply_sequence_command` call, which sets the
+ * sequence space uniquely for each worker.
+ */
+List *
+SequenceDDLCommandsForTable(Oid relationId)
+{
+	List *sequenceDDLList = NIL;
+	List *ownedSequences = getOwnedSequences(relationId);
+	ListCell *listCell;
+	char *ownerName = TableOwner(relationId);
+	bool hasMetadataWorkers = HasMetadataWorkers();
+
+	foreach(listCell, ownedSequences)
+	{
+		Oid sequenceOid = (Oid) lfirst_oid(listCell);
+		char *sequenceDef = pg_get_sequencedef_string(sequenceOid);
+		char *escapedSequenceDef = quote_literal_cstr(sequenceDef);
+		StringInfo wrappedSequenceDef = makeStringInfo();
+		StringInfo sequenceGrantStmt = makeStringInfo();
+		Oid schemaId = InvalidOid;
+		char *createSchemaCommand = NULL;
+		char *sequenceName = get_rel_name(sequenceOid);
+
+		if (hasMetadataWorkers)
```

Comment:
The place of this check was a bit weird. I moved the check at the beginning, since if there are no metadata workers, no need to run the rest of the code. However, I still like this check here because it would require double check (for calls from `start_metadata_sync_to_node` and `create_distributed_table`).